homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -34,15 +39,27 @@
         vol.All(cv.ensure_list, [vol.In(list(SENSORS))])})
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, add_entities,
+                               discovery_info=None):
     """Set up the Enphase Envoy sensor."""
+    from envoy_reader.envoy_reader import EnvoyReader
+
     ip_address = config[CONF_IP_ADDRESS]
     monitored_conditions = config[CONF_MONITORED_CONDITIONS]
 
     # Iterate through the list of sensors
     for condition in monitored_conditions:
-        add_entities([Envoy(ip_address, condition, SENSORS[condition][0],
-                            SENSORS[condition][1])], True)
+        if condition == "inverters":
+            inverters = await EnvoyReader(ip_address).inverters_production()
+            if isinstance(inverters, dict):
+                for inverter in inverters:
+                    add_entities([Envoy(ip_address, condition,
```

Comment:
We should collect all the entities in a list first. Then call `add_entities` once for all entities. I missed it the last review.